source/src/molecular-unfolding/cdk/utils/custom-resource.ts

```diff
@@ -0,0 +1,206 @@
+import * as cdk from 'aws-cdk-lib';
+import {
+  Stack
+} from 'aws-cdk-lib';
+import {
+  Construct
+} from 'constructs'
+
+import {
+  aws_iam as iam
+} from 'aws-cdk-lib'
+
+import {
+  aws_ec2 as ec2
+} from 'aws-cdk-lib'
+
+import {
+  NodejsFunction
+} from 'aws-cdk-lib/aws-lambda-nodejs';
+
+import {
+  Runtime
+} from 'aws-cdk-lib/aws-lambda';
+
+import {
+  Provider
+} from 'aws-cdk-lib/custom-resources';
+
+
+function createCustomResourceLambdaRole(scope: Construct, roleName: string): iam.Role {
+  const account_id = Stack.of(scope).account
+  const region = Stack.of(scope).region
+
+  const role = new iam.Role(scope, `${roleName}`, {
+    assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+  });
+
+  role.addToPolicy(new iam.PolicyStatement({
+    effect: iam.Effect.ALLOW,
+    resources: [
+      '*'
+    ],
+    actions: [
+      "ec2:CreateNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DeleteNetworkInterface",
+      "ec2:AssignPrivateIpAddresses",
+      "ec2:UnassignPrivateIpAddresses"
+    ]
+  }));
+
+  role.addToPolicy(new iam.PolicyStatement({
+    resources: [
+      cdk.Arn.format({
+        service: 'logs',
+        resource: 'log-group',
+        resourceName: '*',
+        arnFormat: cdk.ArnFormat.COLON_RESOURCE_NAME,
+      }, cdk.Stack.of(scope))
+    ],
+    actions: [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:CreateLogGroup"
+    ]
+  }));
+
+  role.addToPolicy(new iam.PolicyStatement({
+    resources: [
+      cdk.Arn.format({
+        service: 'events',
+        region: 'us-west-2',
+        resource: 'rule/*',
```

Comment:
is it necessary to grant the permission to all rules?

Reply:
will add prefix:  `rule/QRSFDD-BraketEventTo${region}*`